tests/test_converter.py

```diff
@@ -227,37 +250,196 @@ def test_parameter_was_not_bound(self, recorder):
             with recorder:
                 quantum_circuit(params={})
 
-    def test_invalid_parameter_expression(self, recorder):
```

Comment:
If you're wondering, I also moved some tests around to try and organize the file better.